src/input.cpp

```diff
@@ -1145,38 +1157,205 @@ std::string input_context::get_desc(
+        inline_fmt = to_translation( "keybinding", string_format( "%s%s%s%s</color>%s%s", "%1$s", lp_sep,
+                                     c_key,
+                                     "%2$s", rp_sep, "%3$s" ) );
+        // \u00A0 is the non-breaking space
+        //~ %1$s: key description,
+        //~ %2$s: action description.
+        separate_fmt = to_translation( "keybinding", string_format( "%s%s%s</color>%s%s", lb_sep, c_key,
+                                       "%1$s",
+                                       rb_sep, "%2$s" ) );
+    }
+    return get_desc( action_descriptor, text, evt_filter, inline_fmt,
+                     separate_fmt );
+}
+
+const nc_color input_context::get_hint_color( keybinding_hint_state state )
+{
+    switch( state ) {
+        case keybinding_hint_state::DISABLED:
+            return c_dark_gray;
+        case keybinding_hint_state::TOGGLED_ON:
+            return c_green;
+        case keybinding_hint_state::TOGGLED_OFF:
+            return c_red;
+        case keybinding_hint_state::HIGHLIGHTED:
+            return h_light_gray;
+        default:
+            return c_light_gray;
+    }
+}
+const nc_color input_context::get_hint_color_for_separator( keybinding_hint_state state )
 {
-    return get_desc( action_descriptor, text, evt_filter,
-                     to_translation(
-                         //~ %1$s: action description text before key,
-                         //~ %2$s: key description,
-                         //~ %3$s: action description text after key.
-                         "keybinding", "%1$s(%2$s)%3$s" ),
-                     to_translation(
-                         // \u00A0 is the non-breaking space
-                         //~ %1$s: key description,
-                         //~ %2$s: action description.
-                         "keybinding", "[%1$s]\u00A0%2$s" ) );
+    switch( state ) {
+        case keybinding_hint_state::DISABLED:
+            return c_dark_gray;
+        case keybinding_hint_state::HIGHLIGHTED:
+            return h_light_gray;
+        default:
+            return c_light_gray;
+    }
 }
 
-std::string input_context::describe_key_and_name( const std::string &action_descriptor,
-        const input_context::input_event_filter &evt_filter ) const
+const nc_color input_context::get_hint_color_for_key( keybinding_hint_state state )
 {
-    return get_desc( action_descriptor, get_action_name( action_descriptor ), evt_filter );
+    switch( state ) {
+        case keybinding_hint_state::DISABLED:
+            return c_dark_gray;
+        case keybinding_hint_state::HIGHLIGHTED:
+            return get_option<bool>( "KEYBINDINGS_ALTERNATE_COLOR_THEME" ) ? h_cyan : h_yellow;
+        default:
+            return get_option<bool>( "KEYBINDINGS_ALTERNATE_COLOR_THEME" ) ? c_cyan : c_yellow;
+    }
 }
 
+std::string replace_spaces_with_non_breaking( std::string s )
+{
+    return std::regex_replace( s, std::regex( " " ), "\u00A0" );
```

Comment:
Use regex for a simple substring replacement might be overkill.

Reply:
The problem I was running into is that `std::replace` doesn't handle strings, it only works with characters which is a problem when dealing with unicode values.

Should I replace this call to `regex_replace` with custom string manipulation code?

Reply:
Huh, didn't know that the standard library does not provide string replacement. Yeah I think a custom string manipulation function would be better here.

Reply:
Done!